### PR TITLE
avoid racyness from the polling library

### PR DESCRIPTION
**Pull request by tgummerer**
The library we're using unfortunately has a race condition that's especially worse when using polling to check for new changes of the file.  Avoid this racyness by sleeping for a bit for now, but really this should be fixed upstream ideally.

I opened a upstream issue for this in https://github.com/nxadm/tail/issues/67.  I think we should merge this in the meantime, as it makes this work, even though it's very hacky. 

I tested this in https://github.com/pulumi/pulumi/pull/15853, where I could no longer reproduce the failure we've seen before.

Fixes https://github.com/pulumi/pulumi/issues/15659
Fixes https://github.com/pulumi/pulumi/issues/15235

### CODE REVIEW

File: sdk/go/auto/stack.go [hunk 1]
```diff
@@ -1455,6 +1456,21 @@ func (fw *fileWatcher) Close() {
...
 
+	// The tail library we're using is racy when shutting down.
+	// If it gets the shutdown signal before reading the data, it
+	// will just shut down before finding the EoF.  This problem
+	// is exacerbated on Windows, where we use the poller, which
+	// polls for changes every 250ms.  Sleep a little bit longer
+	// than that to ensure the tail library had a chance to read
+	// the whole file.  On OSs that don't use the poller we still
+	// want to try to avoid the problem so we sleep for a short
+	// amount of time.
```
Comment (justinvp) [inline sdk/go/auto/stack.go:1467]: Should we include a link to https://github.com/nxadm/tail/issues/67 in the comment?

**Review by justinvp — APPROVED** [review]
Definitely a hack, but seems like a reasonable workaround in the meantime until it's fixed upstream... Would be great if we could submit a PR upstream with a fix.

File: sdk/go/auto/stack.go (hunk 1 above)
Comment (tgummerer) [reply sdk/go/auto/stack.go:1467]: Good idea, done :heavy_check_mark:

**Comment by tgummerer** [comment]
> Would be great if we could submit a PR upstream with a fix.

Yup, I was planning to do that next week after the hackathon is over, since I was going to prioritize the hackathon, but also really wanted to improve the CI situation first.

**Pull request merged**